packages/compass-schema/src/plugin.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { StoreConnector } from '@mongodb-js/compass-components';
```

Comment:
The `LegacyRefluxStoreProvider` that we have in hadron-app-registry is doing the same thing in theory. Is there any particular reason you are keeping usage of this one here?

Reply:
Yeah, thanks for the prod about this :slightly_smiling_face: I just think this line [here](https://github.com/mongodb-js/compass/blob/2ab42e54a31d632e9abee3cebdece6dd51a43152/packages/hadron-app-registry/src/register-plugin.tsx#L61) was buggy, `LegacyRefluxProvider` wouldn’t re-render (and in particular not re-render the actual component) on state changes because the object identity of `state` remained the same?

I pushed 6c58584a3b60c7ad3, that should take care of this for good I think

Reply:
Ohhh, nice catch! Interesting, so class component `setState` and function component `setState` are not the same (because otherwise those are doing the same thing, right? 🤔)

Reply:
class `setState` does partial applications (i.e. `state = { ...state, ...newState }`) and function component `setState` does straight-up assignment (`state = newState`), I think that might be where that’s coming from?

Reply:
Totally! Haven't used class components long enough by now that completely forgot about that 😆